library/src/main/java/com/yayandroid/locationmanager/providers/locationprovider/GooglePlayServicesLocationProvider.java

```diff
@@ -237,14 +262,9 @@ void requestLocationUpdate() {
             getListener().onProcessTypeChanged(ProcessType.GETTING_LOCATION_FROM_GOOGLE_PLAY_SERVICES);
         }
 
-        if (getSourceProvider().isGoogleApiClientConnected()) {
-            LogUtils.logI("Requesting location update...");
-            getSourceProvider().requestLocationUpdate();
-        } else {
-            LogUtils.logI("Tried to requestLocationUpdate, but GoogleApiClient wasn't connected. Trying to connect...");
-            waitingForConnectionToRequestLocationUpdate(true);
-            getSourceProvider().connectGoogleApiClient();
-        }
+        LogUtils.logI("Requesting location update...");
+        setWaiting(true);
```

Comment:
This is set on `get` already, you don't need to set it again. 
Because get is the trigger point, and once this is true - there is either `fail` or `onLocationChanged` calls can happen, which will set to false - and both are terminal. There is no need to set it to true, it will misbehave on the case `keepTracking` 

So, please remove it :)